Add entities Observable for onPageLoad

jweisman opened the pull request:
Following up on #24 , added the `entities$` observable to the `CloudAppEventsService`. 

Usage:
```
<section *ngIf="eventsService.entities$ | async as entities">
  <p *ngFor="let entity of entities">{{entity.description}}</p>
</section>
```

Or:
*component.ts*
```
  items$: Observable<Item[]> = this.eventsService.entities$
  .pipe(
    switchMap((entities) => {
      const items = entities.filter(e=>e.type==EntityType.ITEM);
      return iif(()=>items.length>0,
        forkJoin(items.map(e=>this.restService.call<Item>(e.link))),
        of(null)
      )
    })
  );
```
*component.html*
```
<section *ngIf="items$ | async as items">
  <p *ngFor="let item of items">Barcode-{{item.item_data.barcode}}</p>
</section>
```

jweisman commented:
@gpeterso , @exlbashirn - see the new PR. Happy to receive any comments....

gpeterso reviewed (commented):
@jweisman, from an SDK user's perspective, this seems useful to me. 

I'm pretty indifferent on this point, but based on [comments from @exlbashirn](https://github.com/ExLibrisGroup/cloudapp-sdk/pull/24#issuecomment-751903792), as long as `entities$` is moving to the event service, it might be more straightforward to obtain the observable from the private behavior subject instead of using `fromEventPattern`. For instance:

```typescript
readonly entities$: Observable<Entity[]> = 
    this._onPageLoadSubject$.asObservable().pipe(...);
```

@exlbashirn, did I understand your comment correctly?

exlbashirn commented:
@gpeterso - Yes, that is correct. 
`fromEventPattern` would not be needed in the events service since the subject is directly accessible there.

jweisman commented:
Thanks @exlbashirn - that makes sense. I removed the `fromEventPattern` and referenced the `_onPageLoadSubject$` directly. I added the set to the `init` method. 

Can you take a quick glance?

exlbashirn commented:
@jweisman - Looks good to me.

jweisman commented:
Thanks!

jweisman commented:
Hi @gpeterso - just to let you know this was included in the 0.5.0 version of the SDK, which was released. We also updated the [docs](https://developers.exlibrisgroup.com/cloudapps/docs/api/events-service/#entities) and the [starter app](https://developers.exlibrisgroup.com/cloudapps/tutorials/scaffolding/) to use the new Observable. 

Let us know if you have any comments.

gpeterso commented:
This is great. Thanks @jweisman!
